frontend/mux.go

```diff
@@ -0,0 +1,515 @@
+package frontend
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/Azure/dalec"
+	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/frontend/dockerui"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/moby/buildkit/frontend/subrequests"
+	bktargets "github.com/moby/buildkit/frontend/subrequests/targets"
+	"github.com/moby/buildkit/util/bklog"
+	"github.com/pkg/errors"
+	"golang.org/x/exp/maps"
+)
+
+// BuildMux implements a buildkit BuildFunc via its Handle method. With a
+// BuildMux you register routes with mux.Add("someKey", SomeHandler,
+// optionalTargetInfo).
+//
+// When a build request is made, BuildMux tries to match the requested build
+// target to a registered handler via the registered handler. The correct
+// handler to use is determined with the following logic:
+//
+// Build target is an exact match to a registered route Build target is empty,
```

Comment:
I think the formatting got mangled here. It's hard to understand what this is saying.

Reply:
Fixed